R/assert.R

```diff
@@ -2,13 +2,41 @@
 #'
 #' @description
 #'
-#' * `vec_is()` is a predicate that checks if its input conforms to a
-#'   prototype and/or a size.
-#'
-#' * `vec_assert()` throws an error when the input doesn't conform.
+#' * `vec_is()` is a predicate that checks if its input is a vector that
+#'   conforms to a prototype and/or a size.
+#'
+#' * `vec_assert()` throws an error when the input is not a vector or
+#'   doesn't conform.
+#'
+#' @section Scalars and vectors:
+# From vec_base_typeof():
+#' An object is a vector if one of the following conditions hold:
+#'
+#' - [typeof()] is atomic: `"logical"`, `"integer"`, `"double"`,
+#'   `"complex"`, `"character"`, `"raw"`
+#' - `typeof()` is `"list"`, and one of:
+#'     - the object is a bare `"list"` without a `"class"` attribute,
+#'       see [is_bare_list()]
+#'     - the object is a [data.frame]
```

Comment:
I think it's confusing to have data frames as a sub-bullet of list. Data frames are their own type in vctrs.